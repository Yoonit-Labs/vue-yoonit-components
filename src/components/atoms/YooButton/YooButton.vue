<template lang="pug">
.yoo__container--button
  button(
    :class="['yoo-btn', ...takeModifier]"
    :disabled="disabled"
    @click="$emit('onClick')"
  )
    i(
      v-if="icon.length && iconPosition === 'left'"
      :class="['yoo-btn__icon', takeIconStyle, takeIconName, ...takeModifierIcon]"
      :style="takeIconColor"
    )
    p.yoo-btn__text {{ text }}

    i(
      v-if="icon.length && iconPosition === 'right'"
      :class="['yoo-btn__icon', takeIconStyle, takeIconName, ...takeModifierIcon]"
      :style="takeIconColor"
    )
</template>
<script>
import PropsConfig from '@/components/atoms/YooButton/YooButton.config'
export default {
  name: 'YooButton',
  props: {
    variation: {
      type: String,
      default: 'base',
      validator: value => PropsConfig.variation.options.includes(value)
    },
    buttonSize: {
      type: String,
      default: 'normal',
      validator: value => PropsConfig.buttonSize.options.includes(value)
    },
    fill: {
      type: String,
      default: 'primary',
      validator: value => PropsConfig.fill.options.includes(value)
    },
    icon: {
      type: String,
      default: ''
    },
    iconPosition: {
      type: String,
      default: 'left',
      validator: value => PropsConfig.iconPosition.options.includes(value)
    },
    iconSize: {
      type: String,
      default: 'normal',
      validator: value => PropsConfig.iconSize.options.includes(value)
    },
    iconStyle: {
      type: String,
      default: 'solid',
      validator: value => PropsConfig.iconStyle.options.includes(value)
    },
    animation: {
      type: String,
      default: 'none',
      validator: value => PropsConfig.animation.options.includes(value)
    },
    disabled: {
      type: Boolean,
      default: false
    },
    text: {
      type: String,
      default: ''
    },
    hover: {
      type: Boolean,
      default: false
    },
    active: {
      type: Boolean,
      default: false
    },
    iconCustomColor: {
      type: String,
      default: '',
      validator (value) {
        if (value.includes('#')) {
          const rx = /^#([0-9a-f]{6}|[0-9a-f]{3})$/i
          return value.match(rx)
        } else {
          return value === ''
        }
      }
    }
  },
  computed: {
    takeIconColor () {
      return { color: `${this.iconCustomColor}` }
    },
    takeModifierIcon () {
      const block = 'yoo-btn__icon'
      const classList = []
      classList.push(`${block}--${this.iconSize}`)
      if (this.iconPosition && this.text) {
        classList.push(`${block}--${this.iconPosition}`)
      }
      return classList
    },
    takeModifier () {
      const block = 'yoo-btn'
      const classList = []
      classList.push(`${block}__fill--${this.fill}`)
      classList.push(`${block}--${this.variation}`)
      classList.push(`${block}--${this.buttonSize}`)
      classList.push(`${block}--animation-${this.animation}`)
      if (this.fill && this.hover && !this.disabled) {
        classList.push(`${block}--hover`)
      }
      if (this.fill && this.active && !this.disabled) {
        classList.push(`${block}--active`)
      }
      if (this.fill && this.disabled) {
        classList.push(`${block}--disabled`)
      }
      return classList
    },
    takeIconName () {
      return `fa-${this.icon}`
    },
    takeIconStyle () {
      return this.iconStyle === 'solid'
        ? 'fas' // fontawesome solid
        : 'far' // fontawesome regular
    }
  }
}
</script>

<style src="./YooButton.sass" lang="sass" scoped></style>
