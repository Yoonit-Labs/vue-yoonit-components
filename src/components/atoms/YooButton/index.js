import yooButton from '@/components/atoms/YooButton/YooButton.vue'
export default yooButton
