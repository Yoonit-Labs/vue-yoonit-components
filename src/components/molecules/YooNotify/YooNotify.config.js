module.exports = {
  notifyFill: {
    options: [
      'primary',
      'danger',
      'light',
      'dark'
    ]
  },
  notifyTextColor: {
    options: [
      'light',
      'dark'
    ]
  },
  notifyTextSize: {
    options: [
      'xxxs',
      'xxs',
      'xs',
      'sm',
      'md',
      'lg',
      'xl',
      'xxl',
      'xxxl',
      'display'
    ]
  }
}
