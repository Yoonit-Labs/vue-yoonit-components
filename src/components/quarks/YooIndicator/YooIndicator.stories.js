import yooIndicator from './YooIndicator.vue'
import PropsConfig from './YooIndicator.config'

export default {
  title: 'Yoonit/Indicator',
  component: yooIndicator,
  argTypes: {
    fill: {
      control: {
        type: 'select',
        options: PropsConfig.fill.options
      }
    },
    size: {
      control: {
        type: 'select',
        options: PropsConfig.size.options
      }
    }
  }
}

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { yooIndicator },
  template:
    '<yoo-indicator v-bind="$props"></yoo-indicator>'
})

export const Custom = Template.bind({})
Custom.args = { text: 'Yoo Indicator test component' }

export const Default = Template.bind({})
Default.args = { text: '1' }

export const Primary = Template.bind({})
Primary.args = { text: '1', fill: 'primary' }

export const Light = Template.bind({})
Light.args = { text: '2', fill: 'light' }

export const Dark = Template.bind({})
Dark.args = { text: '10', fill: 'dark' }

export const Neutral = Template.bind({})
Neutral.args = { text: '5', fill: 'neutral' }
