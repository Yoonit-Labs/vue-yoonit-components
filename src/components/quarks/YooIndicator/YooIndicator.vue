<template lang="pug">
  div(
    :class="['yoo-indicator', ...takeModifier]"
  ) {{ text }}
</template>

<script>

export default {
  name: 'YooIndicator',
  props: {
    fill: {
      type: String,
      required: false,
      default: 'danger',
      validator: value =>
        [
          'primary',
          'danger',
          'light',
          'dark',
          'neutral'
        ]
          .includes(value)
    },
    size: {
      type: String,
      required: false,
      default: 'normal',
      validator: value =>
        [
          'small',
          'medium',
          'normal',
          'large'
        ]
          .includes(value)
    },
    animation: {
      type: Boolean,
      required: false,
      default: true
    },
    text: {
      type: String,
      required: false,
      default: ''
    }
  },
  components: {},
  data: () => ({}),
  beforeCreate () {},
  created () {},
  beforeMount () {},
  mounted () {},
  beforeUpdate () {},
  updated () {},
  beforeDestroy () {},
  destroyed () {},
  computed: {
    takeModifier () {
      const classList = [`yoo-indicator--${this.size}`, `yoo-indicator--${this.fill}`]

      if (this.animation) {
        classList
          .push('yoo-indicator--pulse')
      }

      return classList
    }
  },
  methods: {},
  filters: {},
  watch: {}
}
</script>

<style src="./YooIndicator.sass" lang="sass" scoped></style>
