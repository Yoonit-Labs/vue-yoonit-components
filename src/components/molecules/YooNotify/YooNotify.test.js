import { shallowMount } from '@vue/test-utils'
import YooNotify from '@/components/molecules/YooNotify/YooNotify.vue'
import PropsConfig from './YooNotify.config'
import YooButton from '@/components/atoms/Button/Button.vue'
import YooIndicator from '@/components/quarks/YooIndicator/YooIndicator.vue'

const classBlock = 'yoo-notify'
const SlotText = 'Default Slot Text'

const mountComponent = () => {
  return shallowMount(YooNotify, {
    slots: { default: SlotText }
  })
}

describe('YooNotify Component', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mountComponent()
  })

  it('Matches Snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Loads the Component HTML', async () => {
    expect(wrapper.classes('yoo-notify')).toBe(true)
    expect(wrapper.find('.yoo-notify__text').exists()).toBe(true)
    // Test check YooIndicator
    wrapper = shallowMount(YooIndicator)
    await wrapper.vm.$nextTick()
    expect(wrapper.find('.yoo-indicator').exists()).toBe(true)
    // Test check YooButton
    wrapper = shallowMount(YooButton)
    await wrapper.vm.$nextTick()
    expect(wrapper.find('.yoo-btn').exists()).toBe(true)
  })

  describe('Props', () => {
    describe('notifyFill', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.notifyFill.options.includes(YooNotify.props.notifyFill.default)).toBe(true)
      })
      PropsConfig.notifyFill.options.forEach(notifyFill => {
        it(`Includes notifyFill class: .${classBlock}--${notifyFill}`, async () => {
          await wrapper.setProps({ notifyFill })
          expect(wrapper.find(`.${classBlock}--${notifyFill}`).exists()).toBe(true)
        })
      })
    })
    describe('notifyTextColor', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.notifyTextColor.options.includes(YooNotify.props.notifyTextColor.default)).toBe(true)
      })
      PropsConfig.notifyTextColor.options.forEach(notifyTextColor => {
        it(`Includes notifyTextColor class: .${classBlock}__text--${notifyTextColor}`, async () => {
          await wrapper.setProps({ notifyTextColor })
          expect(wrapper.find(`.${classBlock}__text--${notifyTextColor}`).exists()).toBe(true)
        })
      })
    })
    describe('notifyTextSize', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.notifyTextSize.options.includes(YooNotify.props.notifyTextSize.default)).toBe(true)
      })
      PropsConfig.notifyTextSize.options.forEach(notifyTextSize => {
        it(`Includes notifyTextSize class: .${classBlock}__text--${notifyTextSize}`, async () => {
          await wrapper.setProps({ notifyTextSize })
          expect(wrapper.find(`.${classBlock}__text--${notifyTextSize}`).exists()).toBe(true)
        })
      })
    })
  }) // describe Props
})
