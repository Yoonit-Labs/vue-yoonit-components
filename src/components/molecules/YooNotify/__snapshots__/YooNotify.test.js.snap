// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooNotify Component Matches Snapshot 1`] = `
<div class="yoo-grid-layout p__t--xs p__b--xs p__r--s p__l--s yoo-grid-layout--h-space-between yoo-grid-layout--v-center yoo-notify yoo-notify--light" style="grid-template-columns: 30px auto 30px; grid-template-rows: auto; width: 100%; height: auto;">
  <div class="yoo-flex-layout" style="flex-direction: row; flex-wrap: nowrap; justify-content: space-between; align-items: center; align-content: stretch;">
    <div class="yoo-indicator yoo-indicator--normal yoo-indicator--danger yoo-indicator--pulse"></div>
    <p class="m__r--l m__l--l yoo-notify__text yoo-notify__text--dark yoo-notify__text--xs">Yoo-Notify</p>
    <div class="yoo__container--button"><button class="yoo-btn yoo-btn__fill--primary yoo-btn--clear yoo-btn--normal yoo-btn--animation-none"><i class="yoo-btn__icon fas fa-times yoo-btn__icon--normal"></i>
        <p class="yoo-btn__text"></p>
        <!---->
      </button></div>
  </div>
</div>
`;
